### ISSUE TITLE
[Process] Incremental output missing since 3.0.2

### ISSUE THREAD

**Issue by OndraM** [symfony/symfony#17937]
Hi, starting with Symfony 3.0.2, getIncrementalOutput() does not contain the produced incremental output in some circumstances.

I traced the cause down to PR #17423. I also discovered that if you call getOutput() before the getIncerementalOutput(), the later one works properly. I guess this is also the reason why it works in unit tests, as the [getOutput](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Process/Tests/ProcessTest.php#L353) is called there before to detect if some new output was already provided.

The difference is most probably in the readPipes() call inside the getOutput() method. However the fix is a bit beyond my knowledge, so at least there is testcase (part of `ProcessTest::testIncrementalOutput()` test)

This test passes:

```php
// ...
foreach (array('foo', 'bar') as $s) {
    sleep(1);
    $p->getOutput();

    $this->assertSame($s, $p->getIncrementalOutput());

    flock($h, LOCK_UN);
}
// ...
```

This does not: (note missing getOutput() call)

```php
// ...
foreach (array('foo', 'bar') as $s) {
    sleep(1);

    $this->assertSame($s, $p->getIncrementalOutput());

    flock($h, LOCK_UN);
}
// ...
```

Is there anything more I can provide? cc @romainneutron @nicolas-grekas 

Thanks!

**Comment by romainneutron** [comment]
I'm gonna check this, thanks for reporting

**Comment by romainneutron** [comment]
I've submitted a patch in #18023

**Comment by romainneutron** [comment]
Thanks for reporting this !

**Comment by romainneutron** [comment]
@OndraM Patch has been merged in 2.7 
It will soon be available in other branches and it will be released in the upcoming version of each branch.

Thanks again and sorry for the inconvenience
